paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass_tester.cc

```diff
@@ -475,6 +483,122 @@ TEST(CpuQuantizePass, reshapeBetweenNonQuantizedOp) {
+  std::stringstream ss;
+  ss << "Quantization scale for the variable " << var_without_scale
+     << " is missing.";
+  std::string error_msg = ss.str();
+  bool caught_exception = false;
+  try {
+    PreparePass(&graph, prog, variable_names, &original_nodes_num,
+                &current_nodes_num, var_without_scale);
+  } catch (paddle::platform::EnforceNotMet& error) {
+    caught_exception = true;
+    std::string ex_msg = error.what();
+    EXPECT_TRUE(ex_msg.find(error_msg) != std::string::npos);
+  }
+  EXPECT_TRUE(caught_exception);
+}
+
+// (a, w)->Conv->o,
+ProgramDesc BuildProgramDescCheckScalesConv() {
+  ProgramDesc prog;
+  SetOp(&prog, "conv2d", "Conv", {"a", "w"}, {"o"}, true, true);
+  return prog;
+}
+
+// (a, w, b)->Fc->o,
+ProgramDesc BuildProgramDescCheckScalesFc() {
+  ProgramDesc prog;
+  SetOp(&prog, "fc", "Fc", {"a", "w", "b"}, {"o"}, true, true);
+  return prog;
+}
+
+// a->Pool->o,
+ProgramDesc BuildProgramDescCheckScalesPool() {
+  ProgramDesc prog;
+  SetOp(&prog, "pool2d", "Pool", {"a"}, {"o"}, true, true);
+  return prog;
+}
+
+// a->Transpose->o,
+ProgramDesc BuildProgramDescCheckScalesTranspose() {
+  ProgramDesc prog;
+  SetOp(&prog, "transpose2", "Transpose", {"a"}, {"o"}, true, true);
+  return prog;
+}
+
+// a->Reshape->o,
+ProgramDesc BuildProgramDescCheckScalesReshape() {
+  ProgramDesc prog;
+  SetOp(&prog, "reshape2", "Reshape", {"a"}, {"o"}, true, true);
+  return prog;
+}
+
+// (a, b)->Concat->o
+ProgramDesc BuildProgramDescCheckScalesConcat() {
+  ProgramDesc prog;
+  SetOp(&prog, "concat", "Concat", {"a", "b"}, {"o"}, true, true);
+  return prog;
+}
+
+// a->PriorBox->o
+ProgramDesc BuildProgramDescCheckScalesPriorBox() {
+  ProgramDesc prog;
+  SetOp(&prog, "prior_box", "PriorBox", {"a", "b"}, {"o", "v"}, true, true);
+  return prog;
+}
+
+TEST(CPUQuantizePass, check_scales_conv) {
+  const std::initializer_list<std::string> var_names = {"a", "w", "o"};
+  MainTestCheckScales(BuildProgramDescCheckScalesConv(), var_names, "a");
+  MainTestCheckScales(BuildProgramDescCheckScalesConv(), var_names, "w");
+  MainTestCheckScales(BuildProgramDescCheckScalesConv(), var_names, "o");
+}
+
+TEST(CPUQuantizePass, check_scales_fc) {
+  const std::initializer_list<std::string> var_names = {"a", "w", "b", "o"};
+  MainTestCheckScales(BuildProgramDescCheckScalesFc(), var_names, "a");
+  MainTestCheckScales(BuildProgramDescCheckScalesFc(), var_names, "w");
+  MainTestCheckScales(BuildProgramDescCheckScalesFc(), var_names, "o");
+}
+
+TEST(CPUQuantizePass, check_scales_pool) {
+  const std::initializer_list<std::string> var_names = {"a", "o"};
+  MainTestCheckScales(BuildProgramDescCheckScalesPool(), var_names, "a");
+  MainTestCheckScales(BuildProgramDescCheckScalesPool(), var_names, "o");
+}
+
+TEST(CPUQuantizePass, check_scales_transpose) {
+  const std::initializer_list<std::string> var_names = {"a", "o"};
+  MainTestCheckScales(BuildProgramDescCheckScalesPool(), var_names, "a");
```

Comment:
This should be `BuildProgramDescCheckScalesTranspose()` and there are more like this below

Reply:
Not needed anymore.